### ISSUE TITLE
Upgrade bundled cherrypy

### ISSUE THREAD

**Issue by matthewrmshin** [cylc/cylc#2113]
We need to upgrade [cherrypy](https://pypi.python.org/pypi/CherryPy) (from modified 3.2 to 10) ~and [jinja2](http://jinja.pocoo.org/) (from 2.8 to 2.9)~.

(*cherrypy* is going to be more challenging as it has been modified in several changesets.)

**Comment by matthewrmshin** [comment]
Version number of cherrypy is going up like crazy in the last year (>30 releases since June 2016 and >10 releases since Boxing Day 2016!) It is now impossible to determine what is the stable release - with 2 major releases within 2 days - only a few days ago. The project is also being split into 3 parts, so it may become a pain to bundle in the future. If situation does not improve, we may need to consider moving to another web framework, e.g. [flask](https://github.com/pallets/flask).

**Comment by hjoliver** [comment]
The Jinja2 upgrade is somewhat urgent, in that users (in a suite portability context) have run into a bug in 2.8 that I have confirmed is fixed in 2.9.   The bug should probably be documented in the Suite Design Guide where use of {% import ... %} is mentioned.  Details, for the record:

Jinja2 variables defined in the top level template (i.e. suite.rc) are available to sub-templates via `{% include ... %}` but not via `{% import ... %}` unless `with context` is added to the import statement.

E.g. if `sub.rc` is:

```
{% set RESULT = STUFF | default("ERROR") %}
```
then the following will print `# ERROR` under jinja2 v 2.8 and (correctly) `# stuff` at v 2.9:

```
{% set STUFF = "stuff" %}
{% from 'sub.rc' import RESULT with context %}
# {{RESULT}}
```

As a workaround I've found the following does work at 2.8:
```
{% set STUFF = "stuff" %}
{% import 'sub.rc' as SUB with context %}
# {{SUB.RESULT}}
```

**Comment by matthewrmshin** [comment]
Superseded by #2563.
